### ISSUE TITLE
Setting overflow to "hidden" isn't good enough, propose that hidden slides are not rendered to DOM (lazy load to DOM)

### ISSUE THREAD

**Issue by jflayhart** [FormidableLabs/nuka-carousel#368]
Hi, we use [Segment](https://segment.com/) for our analytics data and it looks at elements that are visible to the user and sends something like "user viewed image." Said images are part of this carousel and we have found that even though an image is NOT visible to a user on the UI, due to [the following code](https://github.com/FormidableLabs/nuka-carousel/blob/master/src/index.js#L819), our events tracker still crawls through the page and says "oh, slide 6 is visible to me! must be visible to user." Our `slidesToShow` setting is set to 5... What's more interesting is that if I put a breakpoint and step through carousel code, then all images in the carousel are "seen" by the analytics event! It seems the slower it takes for these images to populate in the slide, then they are technically "visible" to the viewport although hidden in the overflow.

All that said, I played around with your library with `display: none` on the images and that seems to truly hide slides that aren't in view from firing off an analytics event.

**tl;dr**
Even though the images are [hidden behind the CSS `overflow` property](https://github.com/FormidableLabs/nuka-carousel/blob/master/src/index.js#L819), they're still "visible" to [Segment's](https://segment.com/) code.

Any idea on how this carousel can really hide images if they are not in view? I think lazy loading might be good here... not to even render a hidden slide to the DOM until "next" is clicked.

We are running nuka-carousel `4.2.1`

**Comment by samwhale** [comment]
Hi @jflayhart!

This is a very interesting problem. It seems to make sense that you're running into [this](https://stackoverflow.com/a/132679/7174587) as `overflow: hidden` can be used to reveal more or less of an element whereas `display: none` removes the element from the DOM.

Did setting the `display: none` mess with the animations when shifting between slides?

**Comment by samwhale** [comment]
I'm not convinced that nuka-carousel needs this to change. What specifically would the benefit to nuka-carousel be by changing from `overflow: hidden` to `display: none` other than catering to this specific testing software?

cc @kenwheeler @ryan-roemer

**Comment by jflayhart** [comment]
@samwhale yeh you're probably right. Good to be aware of I guess, but I didn't mess with animations using `display: none`

Feel free to close.

**Comment by samwhale** [comment]
@jflayhart I'm going to close this. If you have another issue in the future, please don't hesitate to open another ticket if appropriate.

**Comment by dborilovic7** [comment]
5 years later I'm also having trouble with this at work. Keyboard navigation messes up the carousel + we're also having WCAG issues
